lib/src/visitor/async_evaluate.dart

```diff
@@ -66,12 +66,28 @@ Future<EvaluateResult> evaluateAsync(Stylesheet stylesheet,
             sourceMap: sourceMap)
         .run(stylesheet);
 
-/// Evaluates a single [expression] in [environment]
-Future<Value> evaluateExpressionAsync(
-    Expression expression, AsyncEnvironment environment) {
-  var evaluator = new _EvaluateVisitor();
-  return evaluator._withEnvironment(
-      environment, () => expression.accept(evaluator));
+/// Evaluates a single [expression]
+///
+/// The [variables] are available when evaluating [expression]
+///
+/// The [functions] are available as global functions when evaluating
+/// [expression].
+///
+/// Warnings are emitted using [logger], or printed to standard error by
+/// default.
+///
+/// Throws a [SassRuntimeException] if evaluation fails.
+Future<Value> evaluateExpressionAsync(Expression expression,
+    {Map<String, Value> variables,
+    Iterable<AsyncCallable> functions,
+    Logger logger}) {
+  var evaluator = new _EvaluateVisitor(
+      functions: functions, logger: logger, sourceMap: false);
+  for (var name in variables.keys) {
+    evaluator._environment
+        .setVariable(name, variables[name], null, global: true);
+  }
```

Comment:
Nit, but I'd pass in `variables` to `new EvaluateVisitor()` rather than modifying a private variable. I find it's usually cleaner to avoid accessing another class's privates when possible even if they are visible.